### PR TITLE
sample: sha256 add use of PSA multi-part API's

### PR DESCRIPTION
**Pull request by mswarowsky**
Adds how to perform a hasing operation usinghte PSA multipart API's.

Signed-off-by: Markus Swarowsky <markus.swarowsky@nordicsemi.no>

### CODE REVIEW

File: samples/crypto/sha256/src/main.c [hunk 1]
```diff
@@ -73,7 +78,53 @@ int hash_sha256(void)
...
+{
+
+	uint32_t olen;
+	psa_status_t status;
+	uint8_t *input_ptr = m_plain_text;
+	psa_hash_operation_t hash_operation = {0};
+	size_t rest_length_to_hash = sizeof(m_plain_text);
+
+
+	LOG_INF("Hashing using multi-part SHA256...");
```
Comment (SebastianBoe) [inline samples/crypto/sha256/src/main.c:96]:
```suggestion
	LOG_INF("Hashing using multi-part SHA256...");
```

File: samples/crypto/sha256/src/main.c [hunk 2]
```diff
@@ -73,7 +78,53 @@ int hash_sha256(void)
...
+
+	/* Setup a multipart hash operation */
+	status = psa_hash_setup(&hash_operation, PSA_ALG_SHA_256);
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("Could not setup the hash operation! Error %d", status);
+	}
```
Comment (SebastianBoe) [inline samples/crypto/sha256/src/main.c:102]:
```suggestion
		LOG_ERR("Could not setup the hash operation! Error %d", status);
		return APP_ERROR;		
	}
```

**Review by SebastianBoe — APPROVED** [review]

File: samples/crypto/sha256/src/main.c [hunk 3]
```diff
@@ -33,12 +35,15 @@ LOG_MODULE_REGISTER(sha256, LOG_LEVEL_DBG);
...
-#define NRF_CRYPTO_EXAMPLE_SHA256_TEXT_SIZE (100)
+#define NRF_CRYPTO_EXAMPLE_SHA256_TEXT_SIZE (150)
 #define NRF_CRYPTO_EXAMPLE_SHA256_SIZE (32)
+#define NRF_CRYPTO_EXAMPLE_SHA256_CHUNK_SIZE (25)
 
 /* Below text is used as plaintext for computing/verifying the hash. */
 static uint8_t m_plain_text[NRF_CRYPTO_EXAMPLE_SHA256_TEXT_SIZE] = {
 	"Example string to demonstrate basic usage of SHA256."
+	"With a bit longer input stream, that shows that nRF "
+	"Chips can even hash long messages."
```
Comment (Vge0rge) [inline samples/crypto/sha256/src/main.c:46]:
```suggestion
	"Chips can even hash long messages."
```
that's a public sample,  please remove the "can even hash long messages". Make it a bit more boring please.

File: samples/crypto/sha256/src/main.c [hunk 4]
```diff
@@ -73,7 +78,52 @@ int hash_sha256(void)
...
+
+	/* Setup a multipart hash operation */
+	status = psa_hash_setup(&hash_operation, PSA_ALG_SHA_256);
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("Could not setup the hash operation! Error %d", status);
+		return APP_ERROR;
+	}
+
+	/* Feed the chunks of the input data to the PSA driver */
+	while (rest_length_to_hash > 0) {
```
Comment (Vge0rge) [inline samples/crypto/sha256/src/main.c:104]: I am proposing to remove this while all together. Just split the message in two parts and show that. I consider it important for the samples to be very easy to understand. This adds unnecessary complexity in my opinion.

File: samples/crypto/sha256/src/main.c [hunk 5]
```diff
@@ -110,6 +160,8 @@ int main(void)
 		return APP_ERROR;
 	}
 
+	PRINT_HEX("Plaintext to hash", m_plain_text, sizeof(m_plain_text));
+
 	status = hash_sha256();
```
Comment (Vge0rge) [inline samples/crypto/sha256/src/main.c:165]: rename this to hash_singlepart_sha256 for conformity.

**Review by Vge0rge — CHANGES_REQUESTED** [review]
Also the commit messages is a bit off: 
You need to start the commit with "samples: crypto:"
For the commit message "add use"  doesn't sound connect.
You can also omit also PSA I think since all the samples there are PSA samples. 
You can write something like: "Show usage of multi-part hash APIs" or something similar

File: samples/crypto/sha256/src/main.c (hunk 4 above)
Comment (mswarowsky) [reply samples/crypto/sha256/src/main.c:104]: Does make the code look better as error checking is required each time, but can remove the loop

File: samples/crypto/sha256/src/main.c [hunk 6]
```diff
@@ -33,12 +35,15 @@ LOG_MODULE_REGISTER(sha256, LOG_LEVEL_DBG);
 /* ====================================================================== */
 /*				Global variables/defines for the SHA256 example			  */
 
-#define NRF_CRYPTO_EXAMPLE_SHA256_TEXT_SIZE (100)
+#define NRF_CRYPTO_EXAMPLE_SHA256_TEXT_SIZE (150)
 #define NRF_CRYPTO_EXAMPLE_SHA256_SIZE (32)
+#define NRF_CRYPTO_EXAMPLE_SHA256_CHUNK_SIZE (42)
```
Comment (Vge0rge) [inline samples/crypto/sha256/src/main.c:40]: This is unused, either remove or replace the 42 in the code.

**Review by Vge0rge — APPROVED** [review]

**Pull request merged**